src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const meaning in actions) {
+    const point = actions[meaning];
+
+    for (const keys in point) {
```

Comment:
Instead of using an additional cycle, use type checking:
```
if (action.type === 'addProperties') {
 ...
}
```

As I remember, I checked your prev similar work. Try to improve this approach a bit.